internal/services/storage/storage_account_resource.go

```diff
@@ -2149,22 +2149,26 @@ func resourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) err
 		blobClient := storageClient.BlobServicesClient
 		var blobProps storage.BlobServiceProperties
 
-		repeat := true
-		timeout := time.After(5 * time.Minute) // set timeout to 5 minutes
-		for repeat {
-			select {
-			case <-timeout:
-				log.Printf("[INFO] reading blob service properties timeout expired %s", *id)
-				return fmt.Errorf("reading blob service properties for %s: %+v", *id, err)
-			default:
+		// wait for blob service endpoint to become available
+
+		log.Printf("[INFO] reading %s blob service properties properties..", *id)
+		stateConf := &pluginsdk.StateChangeConf{
+			Pending: []string{"NotFound"},
+			Target:  []string{"Available"},
+			Refresh: func() (interface{}, string, error) {
 				blobProps, err = blobClient.GetServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName)
 				if err != nil {
-					log.Printf("[INFO] refreshing blob service properties %s: %+v", *id, err)
-					time.Sleep(5 * time.Second) // wait for 5 seconds before retrying
-				} else {
-					repeat = false // no error, exit loop
+					log.Printf("[INFO] reading %s blob service properties properties: %+v", *id, err)
+					return blobProps, "NotFound", nil
 				}
-			}
+				return blobProps, "Available", nil
+			},
+			MinTimeout: 10 * time.Second,
+			Timeout:    d.Timeout(pluginsdk.TimeoutCreate),
```

Comment:
Shall we use the `pluginsdk.TimeoutRead`?

Reply:
Here i'm not sure. From my point of view i would suggest to keep the `TimeoutCreate` because of it is related to `Create`. In the screenshot above you can see two services (file and static website) used the waiting loop resulting in 2m20s creation time so it is getting closer to the default read timeout 5 min.

Reply:
@magodo any toughts?

Reply:
Yeah, I saw the intent here. I'm fine we keep this as `TimeoutCreate`. I believe once we migrate to using the typed SDK for this resource, we shall be in a better shape about the timeout indication, then.